templates/diskovc/diskovc.py

```diff
@@ -0,0 +1,77 @@
+from js9 import j
+from zerorobot.template.base import TemplateBase
+
+class Diskovc(TemplateBase):
+
+    version = '0.0.1'
+    template_name = "diskovc"
+
+    OVC_TEMPLATE = 'github.com/openvcloud/0-templates/openvcloud/0.0.1'
+    SSH_TEMPLATE = 'github.com/openvcloud/0-templates/sshkey/0.0.1'
+
+    def __init__(self, name, guid=None, data=None):
+        super().__init__(name=name, guid=guid, data=data)
+        
+        self._ovc = None
+    
+    def validate(self):
+        data = self.data
+        # Get object for an OVC service, make sure exactly one is running
+        ovcs = self.api.services.find(template_uid=self.OVC_TEMPLATE, name=data.get('openvcloud', None))
+        if len(ovcs) != 1:
+            raise RuntimeError('found %s openvcloud connections, requires exactly 1' % len(ovcs))
+
+        data['openvcloud'] = ovcs[0].name
+        
+        # ensure uploaded key
+        self.sshkey
+
+        # ensure that disk has a valid type
+        if data['type']  and data['type'].upper() not in ["D", "B"]:
+            raise j.exceptions.Input("diskovc's type must be data (D) or boot (B) only")
+
+        # ensure that limits are given correctly
+        if (data['maxIOPS'] or data['totalIopsSec']) and (data['readIopsSec'] or data['writeIopsSec']):
+            raise j.exceptions.Input("total and read/write of iops_sec cannot be set at the same time")
+
+        if data['totalBytesSec'] and (data['readBytesSec'] or data['writeBytesSec']):
+            raise j.exceptions.Input("total and read/write of bytes_sec cannot be set at the same time")
+
+        if data['totalBytesSecMax'] and (data['readBytesSecMax'] or data.writeBytesSecMax):
+            raise j.exceptions.Input("total and read/write of bytes_sec_max cannot be set at the same time")
+
+        if data['totalIopsSecMax'] and (data['readIopsSecMax'] or data['writeIopsSecMax']):
+            raise j.exceptions.Input("total and read/write of iops_sec_max cannot be set at the same time")
+
+
+    def update_data(self, data):
```

Comment:
If OVC allow it, here you need to deal with changes in the limits.
So if a limit has changed, you need to do a call on OVC so it adapts the limits of the disk to reflect what has been sent in the blueprint